src/App.tsx

```diff
@@ -15,36 +15,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None,
+  Alphabet,
+  Length,
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getPreparedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortField === SortType.Alphabet) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortField === SortType.Length) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const reset = () => {
+    setSortField(SortType.None);
+    setIsReversed(false);
+  };
+
+  const visibleGoods = getPreparedGoods();
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.Alphabet)}
+          type="button"
+          className={`button is-info ${sortField !== SortType.Alphabet ? 'is-light' : ''}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.Length)}
+          type="button"
+          className={`button is-success ${sortField !== SortType.Length ? 'is-light' : ''}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={`button is-warning ${!isReversed ? 'is-light' : ''}`}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== SortType.None || isReversed) && (
```

Comment:
The condition for rendering the `Reset` button is correct, but it might be more readable if you explicitly check for `sortField !== SortType.None` and `isReversed` separately. This is not a critical issue, but improving readability can help with future maintenance.